internal/provider/modifiers.go

```diff
@@ -38,9 +38,8 @@ func (m boolDefaultModifier) Modify(ctx context.Context, req tfsdk.ModifyAttribu
 		return
 	}
 
-	if !bool.Null {
-		return
+	if bool.Unknown {
+		resp.AttributePlan = types.Bool{Value: m.Default}
```

Comment:
so we just set the default vaule when the attribute is unknown ?

Reply:
I would say yes, because thats when you want a default to kick in. In TF you can explicitly set values to `null` as an actual value. When its not present its `unknown`.

Reply:
Ok, makes sense, so I'm assuming we want to allow `null` for the boolean field

Reply:
yeah.